### DISCUSSION TITLE
BlueJay X2: Authentication failed! 4004000000000000

### DISCUSSION THREAD

**Discussion by emp-00** [Navid200/xDrip#169] [Watches]
I am using a BlueJay X2 watch since more than 1 year. Currently running xDrip+ 2021.09.02 and BlueJay Core Module xDrip v2158 -> Collector OFF thus getting BG values from my Samsung Smartphone. CGM: Dexcom G6. I always have my Smartphone in my pocket or within hands-reach nearby and BlueJay is sitting on my wrist.

However, looking at the xDrip+ logs I find **multiple** "BlueJayServiceR: Authentication failed" entries from time to time, see screenshot below. Sometimes the complete log is full of these messages... They appear _at least daily_ and last for several minutes, up to an hour. Then they disappear again until the next occasion. Also my BlueJay sometimes misses a bunch of readings for 15-30 mins - visible as dropouts in the trend graph. So looks like there's some connectivity/authentication issue? What does "40040000xx" mean?

Since there have been no modifications in the xDrip changelogs with respect to BlueJay I assume updating to the latest nighly won't help? What can I do? Maybe I need a BlueJay firmware update, but how can I get this? I already tried rebooting the BlueJay but no improvements. Thanks for suggestions & helping.

![BlueJay-error](https://user-images.githubusercontent.com/60714349/147560152-edc956ca-6661-4031-8803-31768cd55b14.jpg)

**Comment by Navid200** [comment]
I noticed you are using September 2, 2021.  Is there a reason you are not using the current stable release, October 4, 2021?
I'm not suggesting this is the solution.  I'm just curious.

Did this issue start coinciding with you getting your new phone by any chance?

Do you know that there is a Gitter room specific to BlueJay?
https://gitter.im/jamorham/BlueJay

**Reply by emp-00** [reply]
I have a company managed phone and cannot sideload any new xDrip versions -> that's why I need to stick to the September release. :-(

No the issue is not related to the new phone - had exactly the same symptoms on the old Android 10 phone.

Yes I know the BlueJay Gitter room, but since there's almost no traffic and since I never received answers to my questions there, I am giving your discussion forum a chance. Thanks for setting this up, Navid.

**Reply by Navid200** [reply]
Would you mind pointing me to your question in that room?
If you cannot find it, I will.  But, if you have it, it will save me a bit of time.

**Reply by emp-00** [reply]
My Gitter posting is already half a year old --> received no reply (I also posted a screenshot): https://gitter.im/jamorham/BlueJay?at=60cf82ccbcb3e8060ed591ca

**Comment by Navid200** [comment]
@emp-00 I have been told the way to get firmware update is to contact BlueJay customer service on their website.
I believe this is it:
https://bluejay.website/contactus

**Reply by emp-00** [reply]
Thank you very much Navid200 - I have sent them a message hoping for a fix to this annoying problem.

**Comment by Navid200** [comment]
We have discussions now opened in the main repository:
https://github.com/NightscoutFoundation/xDrip/discussions

I am looking for a way to transfer the open discussions from here to there. But, I'm not sure I'll succeed. In that case, it may be easier to open new discussions there.

**Comment by emp-00** [comment]
For the sake of documentation and feedback to other potentially affected users , here's the dialog I had with the BlueJay "tech support". Unfortunately no real workable solution was found :-( -- I have removed names and XXX=me and YYYYY="tech support".

---snip---
Dear XXX

I get the point about the battery life, but still worth checking to test if this eliminates the issue, as this might point towards the cause.

I deal with bluejay issues, and while the watch works with xDrip+, the internals of the app are beyond me.

Yes there might be some issue with the watch, given the intermittent nature of the problem it sounds, as you said, like a "glitch" rather than a failure.

YYYYY

On 05/01/2022 11:41, XXX wrote:
> Dear YYYYY,
>
> Thanks for the recommendations - however, there's a good reason I don't want to run the watch as collector: Battery runtime is less than half the runtime w/o collector. Having only a couple of days uptime is a "no go" for me, unfortunately.
>
> Why am I the only one with these issues? Maybe a hardware glitch? I don't think running the non-X2 firmware makes sense...
>
> What else can we do? Why is there no more detailed info what the cryptic error message actually means? This should be part of your firmware and the xDrip code/documentation - all in your hands, I guess?
>
> Greetings to UK, stay safe & healthy!
>
> XXX
> sent via mobile phone
>
> Jan 5, 2022 12:21:06 YYYY <support@bluejay.website>:
>
>     Dear XXX
>
>     Thanks for this additional good information.
>
>     I don't know what the error messages are that you are getting on the xDrip+ application.
>
>     Looking at the BlueJay configuration, as you have no other devices getting readings, then one thing to try is to set "run collector" to ON, to get direct readings from the sensor on the watch, this might overcome the drop outs that you are getting via the phone.
>
>     If you wish, we could set your watch to upgrade to a later version, with the consideration that this is non standard. Many other X2's are running your current version without the same reports, the update might, or might not, make any difference, and could introduce other unexpected results. If you want to try this let me know.
>
>     I think first try the less invasive method of setting the watch as a collector and see what the results are.
>     YYYYY
>
>     On 04/01/2022 19:06, XXX wrote:
>
>         Answers to your questions:
>
>         - Current phone Samsung Galaxy A52s 5G on Android Android 11
>         - Exactly same issues with my previous phone 2 months ago: Samsung Galaxy A8 on Android 9/10
>         - Yes, the authentication errors are appearing during the dropouts, BUT: seldomly, the error is logged but there's no dropout. Mostly it coincides, though.
>         - Run Collector is always OFF
>         - Run Phone Collector is always ON
>         - No other devices connected to the Dexcom G6 (no pump etc): only my Samsung phone is connected to the transmitter and the phone is connected to the BlueJay watch
>
>         Can you explain what exactly the "Authentication failed 4004000000000..." message means? Is this a Bluetooth issue when the phone is trying to connect to the BlueJay or an xDrip-internal error preventing BT initialization? Why is this error sometimes logged multiply e.g. 5-10 times per minute and the next minute again etc (see my screenshot in Navid200's discussion forum).
>
>
>         On 2022-01-04 14:46, YYYYY wrote:
>
>             Digging deeper into this there are some options and questions
>
>             Your watch is on the current X2 version. There are later versions, which support the GTS and the X2, but these are not in general use with X2's, so the results might vary, and it would be a little experimental. If you want we could set you to take the later version, but beware it is uncommon on the X2.
>
>             Also we should look into your setup on the phone. Please confirm the make and model and android version you are using on the phone ?
>
>             Do the authentication messages in the log correspond with the dropouts ? I am wondering if there is a firm relationship between them, or perhaps there are two unrelated issues, error message and drop outs ?
>
>             In the bluejay watch settings on xDrip+, how is the "run collector" set ? on or off ?
>
>             In the bluejay advanced settings, is "run phone collector" set on or off ?
>
>             Do you have any other devices getting readings, like a pump or other app ? This will clash when >2 devices attempt to get readings, giving drop outs.
>
>             On 03/01/2022 20:10, XXX wrote:
>                 -- *Message : * I have bought and am using an X2 BlueJay with Firmware 2158, Mac adress: 48:59:******:ED. Could you please send me a Firmware update? Reason: I get this error message in the xDrip+ event log and dropout of readings at least twice a day (lasting at least half an hour with no readings): Authentication failed! 4004000000000000, more details see also here: https://github.com/Navid200/xDrip/discussions/169 Navid200 proposed to ask you for a Firmware update -> this is my request. If you have additional ideas please let me know. Many thanks.
>                 *Email :* XXX
>                 *Name :* XXX
---snap---

**Reply by Navid200** [reply]
@emp-00 If you post that in the discussions in the main repository, there may be more people providing support.
